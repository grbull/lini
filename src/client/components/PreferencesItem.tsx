import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';

interface Props {
  children: ReactNode;
  description: string;
  title: string;
}

export function PreferencesItem({
  children,
  description,
  title,
}: Props): ReactElement {
  return (
    <>
      <div className={cn('flex', 'flex-col')}>
        <span className={cn('font-medium')}>{title}:</span>
        <span className={cn('font-light')}>{description}</span>
      </div>
      <span>{children}</span>
    </>
  );
}
