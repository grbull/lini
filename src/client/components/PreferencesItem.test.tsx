/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { PreferencesItem } from './PreferencesItem';

function setup(): RenderResult {
  const utils = render(
    <PreferencesItem description="testing" title="testing">
      Generic child
    </PreferencesItem>
  );

  return { ...utils };
}

describe('PreferencesItem Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
