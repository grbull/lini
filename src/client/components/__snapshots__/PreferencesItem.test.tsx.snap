// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencesItem Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <span
      class="font-medium"
    >
      testing:
    </span>
    <span
      class="font-light"
    >
      testing
    </span>
  </div>
  <span>
    Generic child
  </span>
</DocumentFragment>
`;
